src/HabiticaAPIManager.js

```diff
@@ -177,6 +177,40 @@ class HabiticaAPIManager {
 		return promise;
 	}
 
+  /**
+	 * Make a POST request to the Habitica API.
+	 * Data object returned varies based on the API url called.
+	 * @param {string} baseURL - the url of the api call.
+	 * @param {HabiticaUserID} userID - the ID of the user, needed for authentication.
+	 * @param {HabiticaAPIToken} userAPIToken - the API Token for the user, needed for authentication.
+	 * @param {object} [queryParams={}] - key-value pairs for any parameters needed by the api call.
+	 * @returns {Promise<String>} Promise containing the raw API response data as a string.
+	 */
+	postRequest(baseURL, userID, userAPIToken, queryParams={}) {
+		let promise = new Promise((resolve, reject) => {
+			let req = new XMLHttpRequest();
+			req.open("POST", baseURL);
+
+			req.onerror = function() {
+				reject(this.responseText);
+			};
+
+			req.onload = function() {
+				if (req.status === 201) {
```

Comment:
Unlike with their GET request endpoints, the Habitica API endpoints for POST requests can return a few different success codes (like 200 and 201), so you'll want to make sure your generic postRequest function can handle all of the success codes the API returns.